tests/macsec/test_reboot.py

```diff
@@ -0,0 +1,71 @@
+'''This script is to test the device reboot behavior for SONiC.
+
+Test 1: Line Card
+Step 1: Configure Macsec between neighbor and DUT
+Step 2: Save config on DUT and reboot the line card
+Step 3: Verify macsec connection is re-established after reboot
+
+Test 2: Chassis
+Step 1: Configure Macsec between neighbor and DUT
+Step 2: Save config on DUT and reboot the chassis
+Step 3: Verify macsec connection is re-established after reboot
+
+'''
+
+import logging
+import pytest
+# from time import sleep
+from tests.common import reboot
+from tests.common.helpers.assertions import pytest_assert
+from tests.common.utilities import wait_until
+
+logger = logging.getLogger(__name__)
+
+pytestmark = [
+    pytest.mark.topology('t2')
+]
+
+
+@pytest.mark.reboot
+def test_lc_reboot(duthost, localhost):
+    # duthost = duthosts[enum_rand_one_per_hwsku_hostname]
```

Comment:
Another way here is to use this construct instead of using random_duthost. 
 :https://github.com/sonic-net/sonic-mgmt/blob/master/tests/common/helpers/dut_utils.py#L45
Set one of the linecard in inventory to be macsec_card, this will generate minigraph with required macsec config.

Reply:
I will add a check to ensure a macsec DUT is being used.